modules/programs/nheko.nix

```diff
@@ -0,0 +1,82 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.nheko;
+
+  iniFmt = pkgs.formats.ini { };
+
+  configDir = if pkgs.stdenv.hostPlatform.isDarwin then
+    "Library/Application Support"
+  else
+    config.xdg.configHome;
+
+  attrsKeyToSnakeCase = let
+    camelCaseToSnakeCase =
+      replaceStrings upperChars (map (s: "_${s}") lowerChars);
+
+    f = mapAttrs' (k: v: nameValuePair (camelCaseToSnakeCase k) v);
+  in mapAttrs' (k: v: nameValuePair k (f v));
+in {
+  meta.maintainers = [ hm.maintainers.gvolpe ];
+
+  options.programs.nheko = {
+    enable = mkEnableOption "Qt desktop client for Matrix";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.nheko;
+      example = literalExpression "pkgs.nheko";
+      description = "The nheko package to use";
+    };
+
+    config = mkOption {
+      type = iniFmt.type;
+      default = { };
+      example = literalExpression ''
+        {
+          general.disableCertificateValidation = false;
+          auth = {
+            accessToken = "SECRET";
+            deviceId = "MY_DEVICE";
+            homeServer = "https://matrix-client.matrix.org:443";
+            userId = "@@user:matrix.org";
+          };
+          settings.scaleFactor = 1.0;
+          sidebar.width = 416;
+          user = {
+            alertOnNotification = true;
+            animateImagesOnHover = false;
+            "sidebar\\roomListWidth" = 308;
+          };
+        }
+      '';
+      description = ''
+        Attribute set of Nheko preferences (converted to INI file).
+
+        For now, it is recommended to run nheko and sign-in before filling in the configuration
+        settings in this module, as nheko writes the access token to <filename>$XDG_CONFIG_HOME/nheko/nheko.conf</filename>
+        the first time we sign in, and we need that data into these settings for the correct
+        functionality of the application.
+
+        This a temporary inconvenience, however, as nheko has plans to move the authentication
+        stuff into the local database they currently use. Once this happens, this will no
+        longer be an issue.
+      '';
```

Comment:
```suggestion
      description = ''
        Attribute set of Nheko preferences (converted to INI file).
        </para><para>
        For now, it is recommended to run nheko and sign-in before filling in the configuration
        settings in this module, as nheko writes the access token to <filename>$XDG_CONFIG_HOME/nheko/nheko.conf</filename>
        the first time we sign in, and we need that data into these settings for the correct
        functionality of the application.
        </para><para>
        This a temporary inconvenience, however, as nheko has plans to move the authentication
        stuff into the local database they currently use. Once this happens, this will no
        longer be an issue.
      '';
```

Reply:
Aren't those tags in reverse order? I see a closing tag first, then an opening tag, not sure if that's correct.

Reply:
Added them anyway, as suggested.